src/Talis/Persona/Client/Base.php

```diff
@@ -39,17 +83,21 @@
      * @param array $config An array of options with the following keys: <pre>
      *      persona_host: (string) the persona host you'll be making requests to (e.g. 'http://localhost')
      *      persona_oauth_route: (string) the token api route to query ( e.g: '/oauth/tokens')
-     *      tokencache_redis_host: (string) the host address of redis token cache
-     *      tokencache_redis_port: (integer) the port number the redis host ist listening
-     *      tokencache_redis_db: (integer) the database to connnect to</pre>
      * @param logger the logger to use, otherwise a default will be assigned and used
      * @throws \InvalidArgumentException if any of the required config parameters are missing
      */
-    public function __construct($config,\Psr\Log\LoggerInterface $logger=null) {
+    public function __construct($config, \Psr\Log\LoggerInterface $logger=null, CacheProvider $cacheBackend=null)
```

Comment:
Scrap that.. A setter would be a bad idea as the Guzzle instance is a singleton. If the user attempted to change the backend after the Guzzle instance was created, it would make no difference. The constructor is a sensible place to provide the cache backend.